### PR TITLE
[FIX] Chat Message Reactions REST API End Point

### PR DESCRIPTION
**Pull request by jgtoriginal**
[FIX] Chat Message Reactions REST API End Point

/api/v1/chat.react method is [documented](https://rocket.chat/docs/developer-guides/rest-api/chat/react/) but not present on `packages/rocketchat-api/server/v1/chat.js`

after having a chat with @graywolf336 it seems that he implemented it at some point, but a `git blame` on that file shows no result, hence I just provide an implementation for it since it's very useful to have.

INSTRUCTION: Just follow the [docs](packages/rocketchat-api/server/v1/chat.js)
@RocketChat/core 

NOTE: not on the `rocketchat-reactions` package nor on this endpoint, is there any sort of sanitization in place, so if you send a non existing emoji it would work the same. I'm not sure how to go by that, since we could stick to the default emojis for sanitization, but I'm not sure we could do so with custom ones. But probably that would be an other task centralized on the `rockatchat-reactions` packages that's the one instantiated by this new API method.

### CODE REVIEW

File: packages/rocketchat-api/server/v1/chat.js [hunk 1]
```diff
@@ -255,3 +255,23 @@ RocketChat.API.v1.addRoute('chat.update', { authRequired: true }, {
...
+
+		const msg = RocketChat.models.Messages.findOneById(this.bodyParams.messageId);
+
+		if (!msg) {
+			throw new Meteor.Error('error-message-not-found', 'The provided "messageId" does not match any existing message.');
+		}
+
+		const emoji = this.bodyParams.emoji;
+
+		Meteor.runAsUser(this.userId, () => Meteor.call('setReaction', emoji, msg._id, this.userId));
```
Comment (rodrigok) [inline packages/rocketchat-api/server/v1/chat.js]: Why do you need to pass the `this.userId` if you are already executing the method using runAsUser?

Comment (jgtoriginal) [reply packages/rocketchat-api/server/v1/chat.js]: @rodrigok you are right it is not strictly necessary to do it that way.
However if you look at every single method on that file, they all do it in that very same way.
So I just tried to stick to the standards and to the scope.

Comment (graywolf336) [reply packages/rocketchat-api/server/v1/chat.js]: @jgtoriginal What method(s) and what files are you talking about?

Comment (jgtoriginal) [reply packages/rocketchat-api/server/v1/chat.js]: @graywolf336 every API method on this file `packages/rocketchat-api/server/v1/chat.js` does `Meteor.runAsUser`. Does it make sense?

Comment (graywolf336) [reply packages/rocketchat-api/server/v1/chat.js]: Yes, that part makes sense however the part we are talking about is the passed in `this.userId` to the `Meteor.call('setReaction...` piece of the code along with the new argument `usr` added to the definition of the `setReaction` method. These last two aren't necessary and can be removed.

File: packages/rocketchat-api/server/v1/chat.js [hunk 2]
```diff
@@ -255,3 +255,23 @@ RocketChat.API.v1.addRoute('chat.update', { authRequired: true }, {
...
+
+RocketChat.API.v1.addRoute('chat.react', { authRequired: true }, {
+	post() {
+		if (!this.bodyParams.messageId || !this.bodyParams.messageId.trim()) {
+			throw new Meteor.Error('error-messageid-param-not-provided', 'The required "messageId" param is missing.');
+		}
+
+		const msg = RocketChat.models.Messages.findOneById(this.bodyParams.messageId);
+
+		if (!msg) {
```
Comment (MarcosSpessatto) [inline packages/rocketchat-api/server/v1/chat.js]: You can move this verification to see if have or not message to the setReactions method, cuz already has one call to db for retrieve message, and you can save a call.

Comment (graywolf336) [reply packages/rocketchat-api/server/v1/chat.js]: I'm going to have to disagree with this. That is outside of the scope of this pull request. If we change how the `setReaction` works then we are going to have to change how the other places which calls that method handles the result. The `setReaction` method doesn't return errors when the message doesn't exist and it only returns `false` so I think we should keep this error here for other developer's sake.

Comment (MarcosSpessatto) [reply packages/rocketchat-api/server/v1/chat.js]: Ok @graywolf336, I had not througt about it. You're right.

**Comment by geekgonecrazy** [comment]
Is this one blocked on anything particular?  Would be great to have this land in 0.62.0

**Comment by jgtoriginal** [comment]
@geekgonecrazy sorry, I'm not blocked on this specific task, I just got jammed among other projects. Will get back to this today so as we can proceed. Thanks for the reminder!

**Comment by geekgonecrazy** [comment]
@jgtoriginal no problem :)  Just seems like a great endpoint to make sure lands in 0.62.0.

I believe we intend to release the candidate / feature freeze on the 20th

**Comment by jgtoriginal** [comment]
@MarcosSpessatto I can see that you took over on removing the unnecessary parameter mentioned by @graywolf336 further up.
are you taking over on the end to end testing you mentioned early on?
just for me to chilax and move forward on other task I got in hands. Thx!

**Pull request merged**